src/app/qgsvectorlayerproperties.cpp

```diff
@@ -408,7 +408,11 @@ QgsVectorLayerProperties::QgsVectorLayerProperties(
     mGeometryPrecisionLineEdit->setValidator( new QDoubleValidator( mGeometryPrecisionLineEdit ) );
 
     mRemoveDuplicateNodesCheckbox->setChecked( mLayer->geometryOptions()->removeDuplicateNodes() );
-    mGeometryPrecisionLineEdit->setText( QString::number( mLayer->geometryOptions()->geometryPrecision() ) );
+    bool ok = true;
+    QString precision( QLocale().toString( mLayer->geometryOptions()->geometryPrecision(), ok ) );
+    if ( ! ok )
+      precision = 0.0;
+    mGeometryPrecisionLineEdit->setText( precision );
```

Comment:
Could you add `setText( precision != 0.0 ? precision : QString() )` or equivalent? I did that once, but it got lost along the way somewhere.